protocol/x/clob/keeper/deleveraging.go

```diff
@@ -152,16 +165,22 @@ func (k Keeper) CanDeleverageSubaccount(
 		satypes.Update{SubaccountId: subaccountId},
 	)
 	if err != nil {
-		return false, err
+		return false, false, err
 	}
 
-	// Deleveraging cannot be performed if the subaccounts net collateral is non-negative.
-	if bigNetCollateral.Sign() >= 0 {
-		return false, nil
+	clobPairId, err := k.GetClobPairIdForPerpetual(ctx, perpetualId)
+	if err != nil {
+		return false, false, err
 	}
+	clobPair := k.mustGetClobPair(ctx, clobPairId)
 
-	// The subaccount's total net collateral is negative, so deleveraging can be performed.
-	return true, nil
+	// Negative TNC, deleverage at bankruptcy price.
+	if bigNetCollateral.Sign() == -1 {
+		return true, false, nil
+	}
+
+	// Non-negative TNC, deleverage at oracle price if market is in final settlement.
```

Comment:
suggestion: worth explaining that if a subaccount's TNC is non-negative, then it should always be a valid state transition to deleverage them at the current oracle price?